client/discovery/discovery.js

```diff
@@ -0,0 +1,187 @@
+    "use strict"
+
+    var width,height
+    var chartWidth, chartHeight
+    var margin
+    var svg = d3v4.select("#graph").append("svg")
+    var chartLayer = svg.append("g").classed("chartLayer", true)
+
+    main()
+
+    function main() {
+        var range = 100
+        var data = {
+            // nodes:d3v4.range(0, range).map(function(d){ return {label: "l"+d ,r:~~d3v4.randomUniform(8, 28)()}}),
+            nodes: d3v4.range(0,range).map( (d) => { return {label: "l"+d , r: 10 }}),
+            links:d3v4.range(0, range).map(function(){ return {source:~~d3v4.randomUniform(range)(), target:~~d3v4.randomUniform(range)()} })
+        }
+
+        setSize(data)
+        drawChart(data)
+    }
+
+    function setSize(data) {
+        width = document.querySelector("#graph").clientWidth
+        height = document.querySelector("#graph").clientHeight
+        margin = {top:0, left:0, bottom:0, right:0 }
+
+
+        chartWidth = width - (margin.left+margin.right)
+        chartHeight = height - (margin.top+margin.bottom)
+
+        svg.attr("width", width).attr("height", height)
+
+
+        chartLayer
+            .attr("width", chartWidth)
+            .attr("height", chartHeight)
+            .attr("transform", "translate("+[margin.left, margin.top]+")")
+
+
+    }
+
+    function drawChart(data) {
+
+        var simulation = d3v4.forceSimulation()
+            .force("link", d3v4.forceLink().id(function(d) { return d.index }))
+            .force("collide",d3v4.forceCollide( function(d){return d.r + 8 }).iterations(16) )
+            .force("charge", d3v4.forceManyBody())
+            .force("center", d3v4.forceCenter(chartWidth / 2, chartHeight / 2))
+            .force("y", d3v4.forceY(0))
+            .force("x", d3v4.forceX(0))
+
+        var link = svg.append("g")
+            .attr("class", "links")
+            .selectAll("line")
+            .data(data.links)
+            .enter()
+            .append("line")
+            .attr("stroke", "black")
+
+        var node = svg.append("g")
+            .attr("class", "nodes")
+            .selectAll("circle")
+            .data(data.nodes)
+            .enter().append("circle")
+            .attr("r", function(d){
+              // console.log(d.r, d.index);
+              return d.r
+            })
+            .on("mouseover", mouseOver(.2))
+            .on("mouseout", mouseOut)
+            .call(d3v4.drag()
+                .on("start", dragstarted)
+                .on("drag", dragged)
+                .on("end", dragended));
+
+        // add a label to each node
+        node.append("text")
+            .attr("dx", 14)
+            .attr("dy", ".35em")
+            .text(function(d) {
+                return "hello";
+            })
+            // .style("stroke", "black")
+            // .style("stroke-width", 0.5)
+            // .style("fill", function(d) {
+            //     return #ddd;
+            // });
+
+        var ticked = function() {
```

Comment:
why did you decide to store a reference to a normal anonnymous functions instead of creating a named function or using an arrow function?